ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/PutJobManagement_Test.java

```diff
@@ -536,4 +538,218 @@ public void getCompletedJobs() throws IOException, SignatureException, XmlProces
+                uploadID = multiPartUploadResponse.getInitiateMultipartUploadResult().getUploadId();
+            } catch (final FailedRequestException e) {
+                assertThat(getCacheBytesAvailable(), lessThan(5000000000000L));
+                assertThat(e.getStatusCode(), is(400));
+            }
+            UUID uuid = UUID.randomUUID();
+            if (uploadID != null){
+                uuid = UUID.fromString(uploadID);
+            }
+            final AbortMultiPartUploadResponse abortResponse = client.abortMultiPartUpload(
+                    new AbortMultiPartUploadRequest(BUCKET_NAME, "beowulf", uuid));
+
+            assertThat(abortResponse.getStatusCode(), is(204));
+
+        } catch (final FailedRequestException e) {
+
+            assertThat(e.getStatusCode(), is(404));
+
+        } finally {
+            deleteAllContents(client, BUCKET_NAME);
+        }
+    }
+
+    @Test //TODO expand positive test if test target >5 TB cache is available
+    public void ListMultiPartUploadParts() throws IOException, SignatureException {
+        try {
+           final ListMultiPartUploadPartsResponse response = client.listMultiPartUploadParts(
+                    new ListMultiPartUploadPartsRequest(BUCKET_NAME, "beowulf", UUID.randomUUID()));
+
+            assertThat(response.getStatusCode(), is(200));
+            assertTrue(response.getListPartsResult().getParts().isEmpty());
+
+        } finally {
+            deleteAllContents(client, BUCKET_NAME);
+        }
+    }
+
+    @Test //TODO expand positive test if test target >5 TB cache is available
+    public void CompleteMultiPartUpload() throws IOException, SignatureException {
+        try {
+            client.completeMultiPartUpload(
+                    new CompleteMultiPartUploadRequest(BUCKET_NAME, "beowulf", UUID.randomUUID()));
+
+            fail("Response should have failed because upload part does not exist");
+
+        } catch (final FailedRequestException e) {
+
+            assertThat(e.getStatusCode(), is(404));
+
+        } finally {
+            deleteAllContents(client, BUCKET_NAME);
+        }
+    }
+
+    @Test //TODO expand positive test if test target >5 TB cache is available
+    public void ListMultiPartUploads() throws IOException, SignatureException {
+        try {
+            final ListMultiPartUploadsResponse response = client.listMultiPartUploads(
+                    new ListMultiPartUploadsRequest(BUCKET_NAME));
+
+            assertThat(response.getStatusCode(), is(200));
+            assertTrue(response.getListMultiPartUploadsResult().getUploads().isEmpty());
+
+        } finally {
+            deleteAllContents(client, BUCKET_NAME);
+        }
+    }
+
+    @Ignore("Pending resolution of Content-Length fail in header")
+    @Test //TODO expand positive test if test target >5 TB cache is available
+    public void PutMultiPartUploadPart() throws IOException, SignatureException {
```

Comment:
It seems like we should create a new suite of tests in a different module where we can run these longer running/larger capacity tests.